lua/starfall/libs_sh/mesh.lua

```diff
@@ -762,6 +780,111 @@ if CLIENT then
 		end
 	end
 
+	local meshgenerating = false
+	--- Generates mesh data. If an Mesh object is passed, it will populate that mesh with the data. Otherwise, it will render directly to renderer.
+	-- @param mesh Optional Mesh object, mesh to build. (default: nil)
+	-- @param prim_type Int, primitive type, see MATERIAL
+	-- @param prim_count Int, the amount of primitives
+	-- @param func The function provided that will generate the mesh vertices
+	-- @client
+	function mesh_library.generate(mesh, prim_type, prim_count, func)
+		if meshgenerating then SF.Throw("Dynamic mesh was already started.", 2) end
+		
+		checkpermission(instance, nil, "mesh")
+		
+		checkluatype(prim_type, TYPE_NUMBER)
+		checkluatype(prim_count, TYPE_NUMBER)
+		checkluatype(func, TYPE_FUNCTION)
+		
+		local tri_count = math.ceil(prim_count / 3)
```

Comment:
Primitives doesn't mean vertices, it can also mean other things like quads and polygons.